lib/oauth2/snake_hash.rb

```diff
@@ -0,0 +1,35 @@
+module OAuth2
+  class SnakeHash < ::Hash
```

Comment:
@pboling I tried this approach https://gist.github.com/anvox/c4e2ca33c6fa59d786cf97960157ae9b but it's overkill. Need to rewrite most methods like indifferent hash. I think just a hash which ref snake case key to camel case key is good enough.